plan/common_plans.go

```diff
@@ -407,51 +406,97 @@ type Explain struct {
 	explainedPlans map[int]bool
 }
 
-// prepareExplainInfo4DAGTask generates the following information for every plan:
-// ["id", "parents", "task", "operator info"].
-func (e *Explain) prepareExplainInfo4DAGTask(p PhysicalPlan, taskType string, parentID string) {
-	childrenIDs := make([]string, 0, len(p.Children()))
-	for _, ch := range p.Children() {
-		childrenIDs = append(childrenIDs, ch.ExplainID())
+const (
+	// treeBody indicates the operator tree is not finished.
+	treeBody rune = '│'
+	// treeMiddleNode indicates this operator is in the middle the operator tree.
+	treeMiddleNode rune = '├'
+	// treeLastNode indicates this operator is the last child of the operator tree.
+	treeLastNode rune = '└'
+	// treeGap is used to represents the gap between the branches of the tree.
+	treeGap rune = ' '
+	// treeNodeIdentifier is used to replace the treeGap once we want to attach
+	// a node to a branch.
+	treeNodeIdentifier rune = '─'
+)
+
+func (e *Explain) getIndent4Child(indent string, isLastChild bool) string {
+	if !isLastChild {
+		return string(append([]rune(indent), treeBody, treeGap))
 	}
-	childrenInfo := strings.Join(childrenIDs, ",")
-	operatorInfo := p.ExplainInfo()
-	count := string(strconv.AppendFloat([]byte{}, p.StatsInfo().count, 'f', 2, 64))
-	row := []string{p.ExplainID(), parentID, childrenInfo, taskType, operatorInfo, count}
-	e.Rows = append(e.Rows, row)
+
+	// If the current node is the last node of the current operator tree, we
+	// need to end this sub-tree by changing the closest treeBody to a treeGap.
+	indentBytes := []rune(indent)
+	for i := len(indentBytes) - 1; i >= 0; i-- {
+		if indentBytes[i] == treeBody {
+			indentBytes[i] = treeGap
+			break
+		}
+	}
+
+	// The child belongs to a new sub-tree, which is rooted by this node.
```

Comment:
This comment should not be here. What ever the node is the last child or not. It is always a root of its own subtree.
I think we'd better tell that `We append treeBody and treeGap to indicate that there's a subtree created`. Add add this comment at [here](https://github.com/pingcap/tidb/pull/6894/files#diff-6b4f5c526cce10da4ab025d07a2d0b32R425) too.